xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/NfTokenObject.java

```diff
@@ -0,0 +1,42 @@
+package org.xrpl.xrpl4j.model.client.accounts;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+import java.util.Optional;
+
+/**
+ * Structure of an NFToken stored on the ledger.
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableNfTokenObject.class)
+@JsonDeserialize(as = ImmutableNfTokenObject.class)
+public interface NfTokenObject {
```

Comment:
Can you also add unit tests for this class and make sure that when you deserialize JSON to a `LedgerObject`, the resulting object is a `NfTokenObject`?

Reply:
NfTokenObject is not a LedgerObject, but that applies to NfTokenOfferObject, for which the JSON tests for serializing and deserializing are in there!